lib/types/string/index.js

```diff
@@ -351,45 +354,32 @@ internals.String = class extends Any {
                 let version = guidOptions.version[i];
                 Hoek.assert(typeof version === 'string', 'version at position ' + i + ' must be a string');
                 version = version.toLowerCase();
-                Hoek.assert(uuids[version], 'version at position ' + i + ' must be one of ' + Object.keys(uuids).join(', '));
-                Hoek.assert(versions.indexOf(version) === -1, 'version at position ' + i + ' must not be a duplicate.');
-                versions.push(version);
+                const versionNumber = uuids[version];
+                Hoek.assert(versionNumber, 'version at position ' + i + ' must be one of ' + Object.keys(uuids).join(', '));
+                Hoek.assert(!(versionNumber in versions), 'version at position ' + i + ' must not be a duplicate.');
+
+                versionNumbers.push(versionNumber);
+                versions[versionNumber] = '';
             }
+
+            versionNumbers = versionNumbers.join('');
         }
 
-        const regex = /^([\[{\(]?)([0-9A-F]{8})([:-]?)([0-9A-F]{4})([:-]?)([0-9A-F]{4})([:-]?)([0-9A-F]{4})([:-]?)([0-9A-F]{12})([\]}\)]?)$/i;
+        const guidRegex = new RegExp(`^([\\[{\\(]?)[0-9A-F]{8}([:-]?)[0-9A-F]{4}\\2?[${checkVersion ? versionNumbers : '0-9A-F'}][0-9A-F]{3}\\2?[${checkVersion ? '89AB' : '0-9A-F'}][0-9A-F]{3}\\2?[0-9A-F]{12}([\\]}\\)]?)$`, 'i');
```

Comment:
You and me both, I can't tell you how many times I counted and recounted these things as I was making incremental changes.